### PR TITLE
Added GRS, replica, storage auto-grow, added CRLF to .kql

### PR DESCRIPTION
**Pull request by mspetroula**
# Overview/Summary

Added GRS, replica, storage auto-grow to both PostgreSQL and MySQL added CRLF to .kql because validation was failing in the Lint phase

## Related Issues/Work Items

Replace this with a list of related GitHub Issues and/or ADO Work Items (Internal Only)

- To associate a GitHub Issue, use a [key word](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) preceded with the GitHub issue number.
- To associate an ADO Work Item, use the key word `AB#` succeeded with the [ADO Work Item ID](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## This PR fixes/adds/changes/removes

1.	Modified: azure-resources/DBforMySQL/flexibleServers/recommendations.yaml
2.	Modified: azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml
3.	Added azure-resources/DBforMySQL/flexibleServers/kql/5c96afc3-7d2e-46ff-a4c7-9c32850c441b.kql
4.	Added azure-resources/DBforMySQL/flexibleServers/kql/8176a79d-8645-4e52-96be-a10fc0204fe5.kql
5.	Added azure-resources/DBforMySQL/flexibleServers/kql/b49a8653-cc43-48c9-8513-a2d2e3f14dd1.kql
6.	Added azure-resources/DBforPostgreSQL/flexibleServers/kql/2ab85a67-26be-4ed2-a0bb-101b2513ec63.kql
7.	Added azure-resources/DBforPostgreSQL/flexibleServers/kql/31f4ac4b-29cb-4588-8de2-d8fe6f13ceb3.kql
8.	Added azure-resources/DBforPostgreSQL/flexibleServers/kql/6293a3cc-6b4a-4c0f-9ea7-b8ae8d7dd3d5.kql
9.	Modified modified:   azure-resources/DBforMySQL/flexibleServers/kql/82a9a0f2-24ee-496f-9ad2-25f81710942d.kql
10.	Modified:   azure-resources/DBforMySQL/flexibleServers/kql/88856605-53d8-4bbd-a75b-4a7b14939d32.kql        
11.	Modified:   azure-resources/DBforPostgreSQL/flexibleServers/kql/b2bad57d-7e03-4c0f-9024-597c9eb295bb.kql
12.	 Modified:   azure-resources/DBforPostgreSQL/flexibleServers/kql/ca87914f-aac4-4783-ab67-82a6f936f194.kql

### Breaking Changes

1. _Replace me_
2. _Replace me_

## As part of this pull request I have

- [ x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [ x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [ x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [x ] Ensured PR tests are passing
- [x ] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries

image

image
image

image
image
image

image

image
image
image

image

image

image

image

image

image

image

image

image

image

image

image

image

image

- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: azure-resources/DBforPostgreSQL/flexibleServers/kql/6293a3cc-6b4a-4c0f-9ea7-b8ae8d7dd3d5.kql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Azure Resource Graph Query
```
Comment (ejhenry) [inline azure-resources/DBforPostgreSQL/flexibleServers/kql/6293a3cc-6b4a-4c0f-9ea7-b8ae8d7dd3d5.kql:1]:
```suggestion
// cannot-be-validated-with-arg
```

File: azure-resources/DBforPostgreSQL/flexibleServers/kql/6293a3cc-6b4a-4c0f-9ea7-b8ae8d7dd3d5.kql [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+// Find Database for PostgreSQL instances that do not have a storage auto-grow
```
Comment (ejhenry) [inline azure-resources/DBforPostgreSQL/flexibleServers/kql/6293a3cc-6b4a-4c0f-9ea7-b8ae8d7dd3d5.kql:2]:
```suggestion

```

File: azure-resources/DBforPostgreSQL/flexibleServers/kql/6293a3cc-6b4a-4c0f-9ea7-b8ae8d7dd3d5.kql [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+// auto-grow property not available today in the output
```
Comment (ejhenry) [inline azure-resources/DBforPostgreSQL/flexibleServers/kql/6293a3cc-6b4a-4c0f-9ea7-b8ae8d7dd3d5.kql:3]:
```suggestion

```

File: azure-resources/DBforPostgreSQL/flexibleServers/kql/6293a3cc-6b4a-4c0f-9ea7-b8ae8d7dd3d5.kql [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+// No query available at the moment
```
Comment (ejhenry) [inline azure-resources/DBforPostgreSQL/flexibleServers/kql/6293a3cc-6b4a-4c0f-9ea7-b8ae8d7dd3d5.kql:4]:
```suggestion

```

File: azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml [hunk 5]
```diff
@@ -35,3 +35,60 @@
...
+  recommendationImpact: High
+  recommendationResourceType: Microsoft.DBforPostgreSQL/flexibleServers
+  recommendationMetadataState: Active
+  longDescription: |
+    Configure storage auto-grow to prevent the server from running out of storage and becoming read-only.
+  potentialBenefits: Scale storage automatically to accomodate increasing workloads
+  pgVerified: false
+  publishedToLearn: false
+  publishedToAdvisor: false
+  automationAvailable: arg
```
Comment (ejhenry) [inline azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml:90]:
```suggestion
  automationAvailable: no
```

**Review by ejhenry — COMMENTED** [review]
Thanks for the contribution @mspetroula. Just a few minor edits.

File: azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml [hunk 6]
```diff
@@ -35,3 +35,60 @@
...
+    Configure storage auto-grow to prevent the server from running out of storage and becoming read-only.
+  potentialBenefits: Scale storage automatically to accomodate increasing workloads
+  pgVerified: false
+  publishedToLearn: false
+  publishedToAdvisor: false
+  automationAvailable: no
+  tags: null
+  learnMoreLink:
+    - name: Storage autogrow using Azure portal in Azure Database for PostgreSQL - Flexible Server
+      url: "https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-auto-grow-storage-portal"
```
Comment (ejhenry) [inline azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml:94]:
```suggestion
      url: "https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-auto-grow-storage-portal"
      
```

File: azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml [hunk 7]
```diff
@@ -35,3 +35,60 @@
...
+- description: Configure storage auto-grow
+  aprlGuid: 6293a3cc-6b4a-4c0f-9ea7-b8ae8d7dd3d5
+  recommendationTypeId: null
+  recommendationControl: Scalability
+  recommendationImpact: High
+  recommendationResourceType: Microsoft.DBforPostgreSQL/flexibleServers
+  recommendationMetadataState: Active
+  longDescription: |
+    Configure storage auto-grow to prevent the server from running out of storage and becoming read-only.
+  potentialBenefits: Scale storage automatically to accomodate increasing workloads
```
Comment (ejhenry) [inline azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml:86]:
```suggestion
  potentialBenefits: Scale storage automatically to meet increasing demand
```

File: azure-resources/DBforMySQL/flexibleServers/recommendations.yaml [hunk 8]
```diff
@@ -35,3 +35,60 @@
...
+- description: Configure storage auto-grow
+  aprlGuid: 8176a79d-8645-4e52-96be-a10fc0204fe5
+  recommendationTypeId: null
+  recommendationControl: Scalability
+  recommendationImpact: High
+  recommendationResourceType: Microsoft.DBforMySQL/flexibleServers
+  recommendationMetadataState: Active
+  longDescription: |
+    Configure storage auto-grow to prevent the server from running out of storage and becoming read-only.
+  potentialBenefits: Scale storage automatically to accomodate increasing workloads
```
Comment (ejhenry) [inline azure-resources/DBforMySQL/flexibleServers/recommendations.yaml:86]:
```suggestion
  potentialBenefits: Scale storage automatically to meet increasing demand
```

**Review by ejhenry — CHANGES_REQUESTED** [review]
Edits to potentialBenefits to meet 60 character limit requirement

File: azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml [hunk 9]
```diff
@@ -35,3 +35,61 @@
...
+  pgVerified: false
+  publishedToLearn: false
+  publishedToAdvisor: false
+  automationAvailable: no
+  tags: null
+  learnMoreLink:
+    - name: Storage autogrow using Azure portal in Azure Database for PostgreSQL - Flexible Server
+      url: "https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-auto-grow-storage-portal"
+
```
Comment (ejhenry) [inline azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml:95]:
```suggestion

```

File: azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml [hunk 10]
```diff
@@ -35,3 +35,60 @@
...
+    Configure storage auto-grow to prevent the server from running out of storage and becoming read-only.
+  potentialBenefits: Scale storage automatically to meet increasing demand
+  pgVerified: false
+  publishedToLearn: false
+  publishedToAdvisor: false
+  automationAvailable: no
+  tags: null
+  learnMoreLink:
+    - name: Storage autogrow using Azure portal in Azure Database for PostgreSQL - Flexible Server
+      url: "https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-auto-grow-storage-portal"
```
Comment (ejhenry) [inline azure-resources/DBforPostgreSQL/flexibleServers/recommendations.yaml:94]:
```suggestion
      url: "https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/how-to-auto-grow-storage-portal"

```

**Review by ejhenry — APPROVED** [review]
LGTM

**Review by oZakari — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**